sam_internal.h

```diff
@@ -0,0 +1,37 @@
+/*  sam_internal.h -- internal functions; not part of the public API.
+
+    Copyright (C) 2019 Genome Research Ltd.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.  */
+
+#ifndef HTSLIB_SAM_INTERNAL_H
+#define HTSLIB_SAM_INTERNAL_H
+
+#include "htslib/sam.h"
+
+// Used internally in the SAM format multi-threading.
+int sam_state_destroy(samFile *fp);
+int sam_set_thread_pool(htsFile *fp, htsThreadPool *p);
+int sam_set_threads(htsFile *fp, int nthreads);
+
+#ifdef __cplusplus
+}
```

Comment:
This is missing its `extern "C" {` companion :smile:

Reply:
Thanks.  Strictly I don't think it's needed given that this is an internal header and trying to compile the files that use it with a c++ is likely to be a heroic exercise, but I've added it for completeness.

Reply:
That's true too. I checked the other *_internal.h headers before deciding between suggesting adding `extern "C" {` or removing this one…